Add draggy number input

g10ria opened the pull request:
Component takes a speed option, set to an integer for integer input and a real number for float input

Commit:
Added dragging number option

Commit:
Merge & resolve conflicts

Commit:
Linting

132ikl commented:
Can you make these changes:
* Fix the arrows
* Disable dragging when in manual edit mode
* Hide the default number-type input arrows on Firefox (https://ikl.sh/taCd)
* Separate speed and type control (just add a "float" parameter with boolean value)
* Rename speed to step for clarity

132ikl reviewed (changes requested):
Can you make these changes:
* Fix the arrows
* Disable dragging when in manual edit mode
* Hide the default number-type input arrows on Firefox (https://ikl.sh/taCd)
* Separate speed and type control (just add a "float" parameter with boolean value)
* Rename speed to step for clarity

g10ria commented:
What do you mean by fix the arrows?

132ikl commented:
The increment/decrement arrows on the left/right don't work for me on Chrome or Firefox.

g10ria commented:
Oh huhh that's weird. Ok I'll look into it

Commit:
Make requested changes: fix side arrows; disable dragging in manual edit
mode; hide firefox default input arrows; separate speed and type
control; add option to specify displayed float precision; rename speed
to step

Commit:
Separate DragNumberOption into DragIntegerOption and DragFloatOption

132ikl reviewed (changes requested):
Can you make sure that the pipeline update event isn't sent until the user releases dragging? Currently each step sends an update to the backend.

Commit:
Stop dragging from firing requests to backend on every change

carbotaniuman commented on src/components/baklava-options-plugin/DragFloatOption.vue:
```diff
@@ -0,0 +1,21 @@
+ * floatPrecision - precision to which the float should be shown
+ *      defaults to 3
+ */
+export default class DragIntegerOption extends DragNumberOption {
```
I think you need an `@Component` here.

carbotaniuman commented on src/components/baklava-options-plugin/DragIntegerOption.vue:
```diff
@@ -0,0 +1,18 @@
+ * step - how 'fast' the value is decreased/increased when sliding or using the arrow keys
+ *      defaults to 1
+ */
+export default class DragIntegerOption extends DragNumberOption {
```
I think you need an `@Component` here.

carbotaniuman commented on src/components/baklava-options-plugin/DragNumberOption.vue:
```diff
@@ -16,46 +14,75 @@
+  }
+
+  startDrag() {
+    const moveListener = this.handleMouseMove.bind(this);
```
Why not an arrow function?

carbotaniuman commented on src/components/baklava-options-plugin/DragNumberOption.vue:
```diff
@@ -16,46 +14,75 @@
+  }
+
+  created() {
+    if (this.option.defaultValue) this.displayValue = this.option.defaultValue;
```
this.value (the prop) gets set with the right value at some point in time. Idk if it's at  `mounted()` or `created()`, but Baklava already has that built in, so let's use it.

Commit:
Fixed default value issue with dragging; added min/maxes

132ikl commented:
Can you do the following in order to fix up the history:
1. Reset to 96a88f7
2. Pull with rebase option
3. Cherry-pick f7d5ad3
4. Force push

132ikl reviewed (changes requested):
Can you change it to make sure that if the min/max is not set then it is not limited?

Commit:
Modify clamping so there's no default bounds

Commit:
Lint + minor fix

Commit:
Remove unused imports (oops)

carbotaniuman commented on src/components/baklava-options-plugin/DragIntegerOption.vue:
```diff
@@ -0,0 +1,22 @@
+ * step - how 'fast' the value is decreased/increased when sliding or using the arrow keys
+ *      defaults to 1
+ * min - minimum value
+ *      defaults to -100
```
Outdated.

carbotaniuman commented on src/components/baklava-options-plugin/DragFloatOption.vue:
```diff
@@ -0,0 +1,25 @@
+ *      defaults to 3
+ * min - minimum value
+ *      defaults to -100
+ * max - maximum value
```
Outdated.

carbotaniuman commented on src/components/baklava-options-plugin/DragNumberOption.vue:
```diff
@@ -16,46 +14,84 @@
+
+      // if no change from the mouse events, it was probably a click, so enter edit mode
+      if (originalVal === this.displayValue) this.enterEditMode();
+      if (this instanceof DragIntegerOption) this.tempValue = this.displayValue.toFixed(0);
```
tempValue never gets used anywhere? is this meant to be displayValue?

carbotaniuman replied:
Looks like it's inherited? Just verify its correct.

carbotaniuman commented on src/components/baklava-options-plugin/DragNumberOption.vue:
```diff
@@ -16,46 +14,84 @@
+
+  clamp(num: number) {
+    let ret = num;
+    if (this.option.min) ret = Math.max(this.option.min, ret);
```
test for `=== undefined`, these prevent 0 mins and maxes

carbotaniuman commented on src/components/baklava-options-plugin/DragNumberOption.vue:
```diff
@@ -16,46 +14,84 @@
+  clamp(num: number) {
+    let ret = num;
+    if (this.option.min) ret = Math.max(this.option.min, ret);
+    if (this.option.max) ret = Math.min(this.option.max, ret);
```
test for `=== undefined`, these prevent 0 mins and maxes

Commit:
Make requested changes

carbotaniuman reviewed (approved):
LGTM
